src/MigrationRunner.ts

```diff
@@ -0,0 +1,127 @@
+import { IDatabase } from "pg-promise"
+import { Migration } from "./Migration"
+
+export class MigrationRunner {
+  constructor(
+    protected pgp: IDatabase<{}>,
+    protected migrations: Migration[],
+    protected schemaName: string = "public",
+  ) {
+    const migrationNames = migrations.map((f) => f.name)
+    const nameDups = this.findDups(migrationNames)
+    if (nameDups.length > 0) {
+      throw Error(`Migrations named ${nameDups.join(", ")} are non-unique.`)
+    }
+  }
+
+  public async setup() {
```

Comment:
If setup is required before anything else can be done, we should have a `throwIfNotSetup` check in each of the other functions that throws a helpful error message to call `setup` first